internal/server/data/provideruser.go

```diff
@@ -164,6 +201,31 @@ func ListProviderUsers(tx ReadTxn, opts ListProviderUsersOptions) ([]models.Prov
 	return result, nil
 }
 
+func PatchProviderUserActiveStatus(tx WriteTxn, providerUser *models.ProviderUser) (*models.ProviderUser, error) {
+	if err := validateProviderUserPatch(providerUser); err != nil {
+		return nil, err
+	}
+	providerUser.LastUpdate = time.Now().UTC()
+
+	pu := (*providerUserTable)(providerUser)
+	query := querybuilder.New("UPDATE")
+	query.B(pu.Table())
+	query.B("SET")
+	query.B("active = ?, last_update = ?", providerUser.Active, providerUser.LastUpdate)
+	query.B("WHERE provider_id = ? AND identity_id = ?", providerUser.ProviderID, providerUser.IdentityID)
+	query.B("RETURNING")
+	query.B(columnsForSelect(pu))
+
+	err := tx.QueryRow(query.String(), query.Args...).Scan(pu.ScanFields()...)
+	if err != nil {
+		if strings.Contains(err.Error(), "no rows in result set") {
```

Comment:
Can we use `handleReadError` to translate the error? If that works maybe we should combine `handlerReadEror` into `handlerError` so that we can use that one function everywhere.